python/pydantic_core/core_schema.py

```diff
@@ -3414,6 +3419,9 @@ def arguments_schema(
         arguments: The arguments to use for the arguments schema
         populate_by_name: Whether to populate by name
         var_args_schema: The variable args schema to use for the arguments schema
+        var_kwargs_mode: The validation mode to use for variadic keyword arguments. If `'single'`, every value of the
+            keyword arguments will be validated against the `var_kwargs_schema` schema. If `'unpacked-typed-dict'`,
+            the `schema` argument must be a [`typed_dict_schema`][pydantic_core.core_schema.typed_dict_schema]
```

Comment:
`uniform` seems good to me 👍

Reply:
I wonder about whether to handle `uniform` via the "extras" of the typed dict schema (if such a thing exists) and have simpler code here (i.e. no need for mode, just always require typed dict schema). Maybe a little breaking, though?

Reply:
There's a `extras_schema` key in the `TypedDictSchema`, but it is not used currently. For Pydantic models, we support:

```python
class Model(BaseModel):
    __pydantic_extra__: dict[str, int]
```

But we don't for typed dicts. I believe it's best to wait for https://peps.python.org/pep-0728/, this will probably completely change how we handle extra keys for typed dicts currently.

Reply:
Sounds good. We should probably start validating extra keys as well, just a side note :).